cpp/src/io/parquet/page_enc.cu

```diff
@@ -1236,22 +1238,23 @@ __global__ void __launch_bounds__(128, 8)
               }
             }
 
-            auto const ret = convert_nanoseconds([&]() {
+            auto const& [last_day_nanos, julian_days] = ([&]() {
```

Comment:
Warning: This returns a reference to temporary vars.
```suggestion
            auto const [last_day_nanos, julian_days] = ([&]() {
```

Reply:
Yikes, this is a good catch.

Reply:
is there any compute-sanitizer check that would catch this?

Reply:
Thinking about it I wonder if the compiler should also have caught it?

Reply:
Actually, I think I misjudged this.
https://godbolt.org/z/r9Ydsaf4q

It's a const reference to the return of the function. It's not like the function is returning a reference to a local temporary.

That said, I'd still avoid the reference. :]

Reply:
Here's the reference (aha!) from scripture: https://en.cppreference.com/w/cpp/language/reference_initialization#Lifetime_of_a_temporary

The short of this is that the following code is valid:
```c++
const std::string& rs = "abc"; // rs refers to temporary copy-initialized from char array
const double& rcd2 = 2;        // rcd2 refers to temporary with value 2.0
int i3 = 2;
double&& rrd3 = i3;            // rrd3 refers to temporary with value 2.0
```
@gerashegalov's code is an extension of this. This makes it technically valid. But I would avoid the phrasing, if only to avoid going back to scripture for justification.